### ISSUE TITLE
Birth of Sky Bot Pathing

### ISSUE THREAD

**Issue by VietCongoRiver** [polarathene/biglobby#53]
In the birth of sky heist, additional bots can't follow you to the ground when you parachute out.  They just climb back into the hatch where you first spawn and stay there for the rest of the heist.

**Comment by polarathene** [comment]
Was this ever an issue in the past? No idea if new updates have since changed how things are done. I've not really played the game or modded for it in over a year. I'll leave the issue open for others and if anyone wants to contribute a fix :)

**Comment by VietCongoRiver** [comment]
Alright, so I found what the issue was. In Birth of Sky when you parachute out, only 3 bots are _teleported_ to ground level, which makes sense because payday is a 4 player co-op game.

**Comment by polarathene** [comment]
> only 3 bots are teleported to ground level, which makes sense because payday is a 4 player co-op game.

It's possibly a mission element script that controls that. Might be possible to fix if someone could locate it and modify it. Did it for a few levels before like Framing Frame day 2, everyone got to be inside the train. I think that one just made sure all players were in train, doing for bots might be harder.

**Comment by Shad0wlife** [comment]
I think I have found the issue's origin:
It is deeper in the game files and not script based I think.
The world.mission xml file from birth of sky contains the 3 fields
```
ai_teleport_001
ai_teleport_002
ai_teleport_003
```
which I guess are the 3 bots being teleported. So fixing this bug would require an addition to the pdmod, but sadly a more intrusive one than the network stuff I think.
Also, it might be required to add more triggers to the map. A nicer way would be to reuse the available triggers if that's possible. But I don't know much about map creation, so somebody else might need to look into this.

**Comment by polarathene** [comment]
@Shad0wlife triggers that are script based are detected by BigLobby when requiring a number of players, in most/all cases I believe it will adapt to something suitable for biglobby(eg only require 4 players in a zone to trigger, or all, I can't recall). This should have been resolved for the heist on day 2 where you needed to go into the train at start and on this birth of sky map at the end with the helicopter.

It may be possible to use those teleport fields you found for additional bots, it's likely a script trigger event, and since it was for bots I probably ignored it. The mod was designed for more players, not bots(that was just a setting offered by request). Feel free to fix it. I'm not sure that adding more teleport fields would necessarily resolve it(depends on the logic that script probably runs using those fields).

The community has plans to fork biglobby afaik and continue to maintain it, they've since found a way to inject xml changes instead of requiring a pdmod which is neat.

**Comment by Shad0wlife** [comment]
Yeah, the problem with these triggers is NOT that they only execute a limited amount of times. There are a limited amount of triggers that execute one time, And I guess Biglobby doesn't do anything about those.
With the xml injection of the SuperBLT based version, there may be better options to fix this, since in the worst case, more triggers or some advanced changes to the 3 triggers available can be done I guess.

**Comment by polarathene** [comment]
In certain cases, I believe biglobby is aware in that it will only require the 3-4 triggers of a regular game to progress(some check waits for all the triggers then compares to amount of players in game, which can be greater than the max it expects, so it would treat it like a 4 player lobby. I know I catered for it with several maps that had problems in the past :)

BigLobby3 has been released now by the community(as you seem to be aware with SuperBLT), link is on this github repo README. Your contributions may be better there as I imagine players will migrate to using that version?

I believe the type of thing your talking about that biglobby handles is [elementareatrigger](https://github.com/polarathene/biglobby/blob/master/biglobby/lua/lib/managers/mission/_elementareatrigger.lua) and [elementfilter](https://github.com/polarathene/biglobby/blob/master/biglobby/lua/lib/managers/mission/_elementfilter.lua), there might be more that biglobby has accounted for. These logic bits are what connect to the level data that you're referring to?

**Comment by Shad0wlife** [comment]
Yes, these elements sound right. I don't have time to go through de decompiled xml again right now, but maxbe a bit later. I will test Biglobby 3 as soon as I have time to, because that will offer a much easier way of xml editing for testing if it works.
